src/defer/index.ts

```diff
@@ -0,0 +1,30 @@
+import { Store, Unit, merge, sample } from 'effector';
+import { combineEvents } from '../combine-events';
+import { not } from '../not';
+
+export interface DeferArgs {
+  clock: Unit<any>;
+  until: Store<boolean>;
+}
+
+export const defer = (args: DeferArgs) => {
+  const { clock, until: condition } = args;
+
+  const calledAfterCondition = sample({
+    clock: clock,
+    filter: condition,
+  });
+
+  const calledBeforeCondition = sample({
+    clock: clock,
+    filter: not(condition),
+  });
+
+  return merge([
```

Comment:
there could be 4 cases:

1. Event called, condition is true --> call event
2. Event called, condition is false --> wait until condition will be true and call event
3. Condition became true, event is not called yet --> wait until event is called
4. Condition became false, event is not called yet --> reset the state and wait both for event and condition to become true